projects/js-packages/publicize-components/src/components/form/index.js

```diff
@@ -194,8 +195,8 @@ export default function PublicizeForm( {
 											false === is_healthy
 										}
 										enabled={ enabled && ! isPublicizeDisabledBySitePlan && false !== is_healthy }
-										key={ id }
-										id={ id }
+										key={ connection_id ? connection_id : id }
+										id={ connection_id ? connection_id : id }
```

Comment:
I've just realised that this could be:
```suggestion
										key={ connection_id ?? id }
										id={ connection_id ?? id }
```